seps/1933-workload-identity-federation.md

```diff
@@ -0,0 +1,228 @@
+4. **Operational Complexity**: Separate credential lifecycles for platform identity and MCP access create coordination challenges and increase operational burden
+
+Workloads in modern deployment environments already possess cryptographically verifiable identity credentials issued by their platforms. The current MCP specification provides no mechanism to leverage these existing identities, forcing operators to implement parallel credential management systems. This SEP addresses this gap by defining how workloads can federate their existing platform identities to access MCP servers through trust relationships configured at the authorization server level.
+
+## Specification
+
+The message flow combines using a JWT as an authorization grant as defined in
+[RFC7523](https://datatracker.ietf.org/doc/html/rfc7523) with automated issuer
+discovery as described in [OpenID Connect Discovery 1.0](https://openid.net/specs/openid-connect-discovery-1_0.html)
+to obtain the [RFC7517](https://datatracker.ietf.org/doc/html/rfc7517) JWK
+Key set which is used to validate the JWT authorization grant.
+
+The complete Workload Identity Federation flow proceeds as follows:
+
+```mermaid
+sequenceDiagram
+    participant Client as MCP Client
+    participant Auth as Authorization<br/>Server
+    participant OIDC as OIDC<br/>Discovery<br/>Endpoint
+    participant JWKS_URI as JWK Set<br/>Location
+    participant Server as MCP Server
+
+    Client->>Auth: 1. Request Access Token
+    Note over Auth: 2. Construct OpenID Provider Discovery Endpoint URL
+    Auth->>OIDC: 3. Request OpenID Provider Configuration Document
+    OIDC->>Auth: 4. Return OpenID Provider Configuration Document
+    Note over Auth: 5. Extract `jwks_uri`
+    Auth->>JWKS_URI: 6. Request JWK Set
+    JWKS_URI->>Auth: 7. Return JWK Set
+    Note over Auth: 8. Validate JWT<br/>Authorization Grant
+    Auth->>Client: 9. Return Access Token
+    Client->>Server: 10. MCP Server Access
+    Note over Server: 11. Validate Access Token
+    Server->>Client: 12. MCP Server Response
+```
+
+### Flow Steps
+
+1. **Request Access Token**: The MCP client requests an access token through a
+   POST request to the authorization server's token endpoint using the JWT Bearer
+   grant type as specified in
+   [RFC 7523 Section 2.1](https://datatracker.ietf.org/doc/html/rfc7523#section-2.1).
+   The request includes:
+   - `grant_type`: `urn:ietf:params:oauth:grant-type:jwt-bearer`
+   - `assertion`: The client's workload JWT credential
+   - `resource`: The canonical URI of the target MCP server (as defined in the
+     baseline MCP Authorization specification)
+
+2. **Construct OpenID Provider Discovery Endpoint URL**: The authorization server extracts the `iss` (issuer)
+   claim from the JWT and determines if it is a trusted issuer. If the issuer is
```

Comment:
We probably want something about what a trusted issuer is, and how an MCP Server's AS might be configured with trusted issuers. I'm ok saying that the process is out of scope of this document, may happen by a variety of mechanisms, etc. But we should probably have at least one example, probably as an example from the motivation section.